Sources/apm-agent-ios/Configuration/CentralConfigFetcher.swift

```diff
@@ -0,0 +1,160 @@
+// Copyright © 2023 Elasticsearch BV
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+import Foundation
+import Logging
+
+class CentralConfigFetcher {
+    
+    static let ETAG_KEY = "elastic.central.config.etag"
+    static let MAXAGE_KEY = "elastic.central.config.maxAge"
+    let fetchQueue = DispatchQueue(label: "co.elastic.centralConfigFetch")
+    let fetchTimer : DispatchSourceTimer
+    let logger : Logger
+
+    let serviceEnvironment : String
+    let serviceName : String
+    let config : AgentConfiguration
+    var task : URLSessionDataTask? = nil
+    let callback : (Data) -> Void
+    
+    var etag : String? {
+        get {
+            UserDefaults.standard.object(forKey: Self.ETAG_KEY) as? String
+        }
+        set(etag) {
+            UserDefaults.standard.setValue(etag, forKey:   Self.ETAG_KEY)
+        }
+    }
+
+    var maxAge : TimeInterval? {
+        get {
+            UserDefaults.standard.object(forKey: Self.MAXAGE_KEY) as? TimeInterval
+        }
+        
+        set(maxAge) {
+            fetchTimer.suspend()
+
+            fetchTimer.schedule(deadline: .now() + (maxAge ?? 60.0), repeating: maxAge ?? 60.0 )
```

Comment:
Just a nitpick, no need to change it, though I'd store the number `60.0` in a variable since it seems like it's being used in many places across this file.